ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/array/ArrayConcat.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package io.confluent.ksql.function.udf.array;
+
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@UdfDescription(
+    name = "array_concat",
+    category = FunctionCategory.ARRAY,
+    description = "Concatenates two arrays, creating an array that contains all the elements"
+        + "in the first array followed by all the elements in the second array."
+        + " Returns NULL if both input arrays are NULL. "
+        + "The two arrays must be of the same type.")
+public class ArrayConcat {
+  @Udf
+  public <T> List<T> concat(
+      @UdfParameter(description = "First array of values") final List<T> left,
+      @UdfParameter(description = "Second array of values") final List<T> right) {
+    if (left == null && right == null) {
+      return null;
+    }
+    final List<T> result = new ArrayList(left.size() + right.size());
```

Comment:
if `left == null || right == null` wouldn't we get an NPE here?

Reply:
I thought so too, but there are unit tests for those cases. Not sure what to make of that.

Reply:
thanks for pointing out. missed that when changing the initialization to the exact number of elements. No idea what's going on with the tests, they should fail here. Will double check.

Reply:
Tests did catch this, sorry for missing to run them on that last change. Update fixes the issue in ArrayConcat.